### DISCUSSION TITLE
Public EJBCA Test Instance for PKIaaS Integration

### DISCUSSION THREAD

**Discussion by CardContact** [Keyfactor/ejbca-ce#685] [Show and tell]
We are working on an EJBCA connector for our [PKI-as-a-Service](http://www.pki-as-a-service.com) Portal that shall allow SmartCard-HSM users to easily request and retrieve certificates from EJBCA.

Because not everyone has an EJBCA cloud instance at hand to do some testing and documentation, I'm wondering if there is a public test instance running somewhere in the cloud that we could use to test and document the connector ?

The basic principle of the integration is, that you initially import the admin client certificate and key into a SmartCard-HSM and then connect via the portal to EJBCA. Using this connection, the EJBCA instance is registered in the portal and the RA can approve and request certificates. Portal users can then request certificates using the normal workflow.

We have done a similar integration with the [DFN-PKI](https://www.dfn.de/en/dienste/security-trust-identity/), which provide a test RA to do the integration. Maybe something similar exists for other EJBCA instances.

**Comment by primetomas** [comment]
I have a publicly available, but private, instance that I give access to for integration. Send me an email and I can give access to it. 

Otherwise it's also easy to spin up an instance of EJBCA Cloud, or just run the container locally. We don't have a Public test instance any longer since it's so easy to spin one up. I use the private one to also test patches for integration, a reason that is useful.

**Comment by CardContact** [comment]
We have released a first version of the [EJBCA Connector](https://www.smartcard-hsm.com/2024/12/17/ejbca-connector.html).

Integration has been tested with @primetomas EJBCA cloud, but we are of course interested in feedback from other users running public test instances.

Issue tracking is available in the [deploy-pki-as-a-service](https://github.com/CardContact/deploy-pki-as-a-service) repo.
